src/Mobs/SnowGolem.cpp

```diff
@@ -38,7 +38,7 @@ void cSnowGolem::Tick(std::chrono::milliseconds a_Dt, cChunk & a_Chunk)
 	}
 	if (IsBiomeNoDownfall(m_World->GetBiomeAt(POSX_TOINT, POSZ_TOINT)))
 	{
-		TakeDamage(*this);
+		TakeDamage(dtAttack, nullptr, GetRawDamageAgainst(*this), GetKnockbackAmountAgainst(*this));
```

Comment:
I don't think `dtAttack` is the right choice, because plugins would expect the Attacker parameter to be valid. Perhaps a new constant `dtEnvironment`?

Reply:
Oh - yeah... that might break some things

Note: when using TakeDamage(*this) the damage type defaults to dtAttack with this as cause (which is not nice.... that lead to the circular relation)

The new constant can be also used for the enderman damage from water

Reply:
Do i need to document this somewhere? the comment says there are some notes about the damage types.

Reply:
The only place you should need to document this is in the enum itself, where there is a comment for each enum value. The API documentation doesn't have an additional note for any of the other damage types, and there isn't a reason dtEnvironment would need it either.

Reply:
So the dtEnvironment isn't added to the API automaticly - do I need to do something. The damage types are exported all at once

Reply:
Yep, you're right. They're all exported at once so once you've added another enum entry, there's nothing more to be done.

Reply:
Actually, I take back that last comment. Although logically you shouldn't need to document the additional entry; because it's a new symbol the build fails; the old dt enumeration values are grandfathered in.

Since you have to document one, you may as well document them all. Add them into the APIDesc.lua file here (making sure to maintain alpha-sorting): https://github.com/cuberite/cuberite/blob/master/Server/Plugins/APIDump/APIDesc.lua#L14207

Reply:
Like i did now?

Reply:
Yes, looks good.